mmv1/products/cloudfunctions2/api.yaml

```diff
@@ -0,0 +1,302 @@
+# Copyright 2017 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: CloudFunctions2
+display_name: Cloud Functions (2nd gen)
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://cloudfunctions.googleapis.com/v2/
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Cloud Functions API
+    url: https://console.cloud.google.com/apis/library/cloudfunctions.googleapis.com/
+objects:
+  - !ruby/object:Api::Resource
+    name: 'CloudFunction'
+    base_url: projects/{{project}}/locations/{{location}}/functions
+    create_verb: :POST
+    description: |
+      A Cloud Function that contains user computation executed in response to an event.
+    collection_url_key: 'functions'
+    update_mask: true
+    async: !ruby/object:Api::OpAsync
+      operation: !ruby/object:Api::OpAsync::Operation
+        path: 'name'
+        base_url: '{{op_id}}'
+        wait_ms: 1000
+      result: !ruby/object:Api::OpAsync::Result
+        path: 'response'
+        resource_inside_response: true
+      status: !ruby/object:Api::OpAsync::Status
+        path: 'done'
+        complete: true 
+        allowed:
+          - True
+          - False
+      error: !ruby/object:Api::OpAsync::Error
+        path: 'error'
+        message: 'message'
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: 'location'
```

Comment:
Given the resource can be updated (it's not `input:true`) I expect that a few fields will need to be marked individually. In particular, "Identity" fields of a resource like `location`/`project`/`name` are generally immutable.

Reply:
Updated